src/Analyzer/Passes/QueryAnalysisPass.cpp

```diff
@@ -2249,7 +2251,13 @@ void QueryAnalyzer::validateJoinTableExpressionWithoutAlias(const QueryTreeNodeP
         return;
 
     auto table_expression_node_type = table_expression_node->getNodeType();
-    if (table_expression_node_type == QueryTreeNodeType::TABLE_FUNCTION ||
+    auto * join_typed = join_node->as<JoinNode>();
+    if (table_expression_node_type == QueryTreeNodeType::QUERY && join_typed->getKind() == JoinKind::Paste)
```

Comment:
Sorry, I already forgot what `table_expression_node` corresponds to. Why do we check only one node while `checkDuplicateTableNamesOrAlias` expects left and right expression to be query ?

Reply:
We check only one node each time because we pass into `validateJoinTableExpressionWithoutAlias` only one argument (either left or right), so we call this function two times for each of the values.